packages/performance/src/resources/trace.ts

```diff
@@ -145,26 +146,39 @@ export class Trace implements PerformanceTrace {
 
   /**
    * Increments a custom metric by a certain number or 1 if number not specified. Will create a new
-   * custom metric if one with the given name does not exist.
+   * custom metric if one with the given name does not exist. The value will be floored down to an
+   * integer.
    * @param counter Name of the custom metric
-   * @param num Increment by value
+   * @param numAsInteger Increment by value
    */
-  incrementMetric(counter: string, num = 1): void {
+  incrementMetric(counter: string, numAsInteger = 1): void {
     if (this.counters[counter] === undefined) {
       this.putMetric(counter, 0);
     }
-    this.counters[counter] += num;
+    const valueAsInteger: number = Math.floor(numAsInteger);
```

Comment:
Can we put the shared part in a function at the end of this file?

Reply:
Good idea. Done.